src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1497 @@
+    }
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"tokenFrom\" or \"chainFrom\" must be non-null");
+    if (!metaObj["amountFrom"].isNull())
+        order.amountToFill = order.amountFrom = AmountFromValue(metaObj["amountFrom"]);
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"amountFrom\" must be non-null");
+    if (!metaObj["orderPrice"].isNull())
+        order.orderPrice = AmountFromValue(metaObj["orderPrice"]);
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"orderPrice\" must be non-null");
+    
+    if (!metaObj["expiry"].isNull())
+        order.expiry = metaObj["expiry"].get_int();
+
+    if (tokenFromSymbol.empty() && order.chain.empty())
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, either \"tokenFrom\" or \"chainFrom\" must be non-null. [tokenFrom,chainTo] or [chainFrom,tokenTo]");
+    if (!tokenFromSymbol.empty() && !tokenToSymbol.empty())
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, \"tokenFrom\" and \"tokenTo\" cannot be set in the same time. [tokenFrom,chainTo] or [chainFrom,tokenTo]");
+    
+    if (!tokenFromSymbol.empty())
+        order.orderType = CICXOrder::TYPE_INTERNAL;
+    else
+        order.orderType = CICXOrder::TYPE_EXTERNAL;
+    
+
+    int targetHeight;
+    {
+        LOCK(cs_main);
+        DCT_ID idToken;
+        std::unique_ptr<CToken> token;
+
+        if (order.orderType == CICXOrder::TYPE_INTERNAL)
+        {
+            token = pcustomcsview->GetTokenGuessId(tokenFromSymbol, idToken);
+            if (!token)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Token %s does not exist!", tokenFromSymbol));
+            order.idToken = idToken;
+
+            CTokenAmount balance = pcustomcsview->GetBalance(order.ownerAddress,idToken);
+            if (balance.nValue < order.amountFrom)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Not enough balance for Token %s on address %s!", token->CreateSymbolKey(order.idToken), ScriptToString(order.ownerAddress)));
+        }
+        else
+        {
+            token = pcustomcsview->GetTokenGuessId(tokenToSymbol, idToken);
+            if (!token)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Token %s does not exist!", tokenToSymbol));
+            order.idToken = idToken;
+        }
+
+        targetHeight = ::ChainActive().Height() + 1;
+    }
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::ICXCreateOrder)
+             << order;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    const auto txVersion = GetTransactionVersion(targetHeight);
+    CMutableTransaction rawTx(txVersion);
+
+    CTransactionRef optAuthTx;
+    std::set<CScript> auths;
```

Comment:
No auth set here for the order, is that correct? I'd expect an address to be set as auth here to allow the auth address to be able to close the order with closeorder with an input that corresponds to it.